duckduckgo/macos-browser
# Move VPN ownership to menu agent

Task/Issue URL: https://app.asana.com/0/72649045549333/1205529334042574/f

BSK PR: https://github.com/duckduckgo/BrowserServicesKit/pull/536
iOS PR: https://github.com/duckduckgo/iOS/pull/2101

## Description

Moves the VPN ownership to the menu agent

## Steps to test this PR

Apologies in advance for such a large PR, but this type of fundamental change is hard to split up further than this.  I'll list the things that should be tested:

- Test enabling the VPN from scratch, with the onboarding steps.
- Test Debug menu > Reset Status
- Test Waitlist onboarding flow.
- Test controlling the VPN using the menu app.
- Test notifications, and sending a test notification.

Some aids:
`launchctl list | grep duck` - to see our launchctl services
`launchctl list remove com.duckduckgo.macos.vpn.debug` to remove the VPN (debug) agent

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
